app/main.py

```diff
@@ -2,5 +2,17 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+
+    all_cache = {}
+
+    def inner(*args) -> int:
+        all_args = []
+        for arg in args:
+            all_args.append(arg)
+        if str(all_args) in all_cache:
+            print("Getting from cache")
+        else:
+            print("Calculating new result")
+            all_cache[str(all_args)] = func(*args)
+        return all_cache[str(all_args)]
+    return inner
```

Comment:
no need to use loop here 

*  **Tuples as keys:**
   - Use tuples directly as dictionary keys instead of strings.

*  **Simplified caching logic:**
   - Check if arguments are in cache; if so, print "Getting from cache" and return cached result.
   - If not, compute result, store in cache, print "Calculating new result", and return result.